### PR TITLE
feat: HorizontalScroll component

### PR DESCRIPTION
**Pull request by mainframev**
I tried to find the most simple solution possible since such components can have different ways of implementation, I tried to avoid approaches with calculations because it will make it more complicated. So the approach I picked was - to hide the scroll bar and handle the dragging movement looks neater, I think. 
 
 Storybook: https://orbit-feat-horizontall-scroll.surge.sh

### CODE REVIEW

File: packages/orbit-components/src/HorizontalScroll/__tests__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-size: 20px;
+  color: #fff;
+  font-weight: bold;
+  width: 150px;
+  background: grey;
+  height: 50px;
+`;
```
Comment (silvenon) [inline packages/orbit-components/src/HorizontalScroll/__tests__/index.test.js:18]: Why style a div in unit tests?

File: packages/orbit-components/src/HorizontalScroll/index.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+const StyledContainer = styled.div`
+  ${({ isDragging }) => `
+    height: 100%;
+    display: inline-flex;
+    pointer-events: ${isDragging && "none"};
+  `}
+`;
+
+const StyledChild = styled.div``;
```
Comment (silvenon) [inline packages/orbit-components/src/HorizontalScroll/index.js:43]: You don't need this.

File: packages/orbit-components/src/HorizontalScroll/HorizontalScroll.stories.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  background: grey;
+  height: 50px;
+`;
+
+export const Default = (): React.Node => {
+  return (
+    <HorizontalScroll>
+      {Array(...Array(10)).map((_, key) => (
+        // eslint-disable-next-line
+        <CustomDiv key={key}>{key}</CustomDiv>
```
Comment (silvenon) [inline packages/orbit-components/src/HorizontalScroll/HorizontalScroll.stories.js:28]: I'd try using a combination of `Box` + `Text` instead, or at least use the sans-serif font.

**Review by silvenon — CHANGES_REQUESTED** [review]
Other than comments below, it seems to work good 👍

File: packages/orbit-components/src/HorizontalScroll/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  </HorizontalScroll>
+  ```
+
+## Props
+
+| Name      | Type                    | Required           | Default | Description                           |
+| --------- | ----------------------- | ------------------ | ------- | ------------------------------------- |
+| minHeight | `number`                |                    |         | set minimal height                    |
+| dataTest  | `string`                |                    |         | prop for testing purposes             |
+| spacing   | [`Spacing`](###Spacing) |                    | "small" | the spacing between children elements |
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/HorizontalScroll/README.md]: what's wrong with `###`? it looks even better, i think :)

File: packages/orbit-components/src/HorizontalScroll/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+export interface Props extends Common.Global {
+  /** set minimal height */
+  readonly minHeight?: number;
+  /** prop for testing purposes */
+  readonly dataTest?: string;
+  /** The spacing between children elements */
+  /** default: "small" */
+  readonly spacing?: Spacing;
+  /** content of HorizontalScroll */
+  readonly children: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/HorizontalScroll/index.d.ts:37]: hm, this component always should have children, so i think it's should be required 🤔

File: packages/orbit-components/src/HorizontalScroll/index.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+const HorizontalScroll = ({
+  children,
+  spacing = "small",
+  dataTest,
+  minHeight,
+  ...props
+}: Props): React.Node => {
+  const [height, setHeight] = React.useState(0);
+  const { current: childRef } = React.useRef<(number | null)[]>([]);
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/HorizontalScroll/index.js:53]: not a mistake, personal preference to have it like that.

File: packages/orbit-components/src/HorizontalScroll/index.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      data-test={dataTest}
+    >
+      <StyledOverflow ref={scrollWrapper}>
+        <StyledContainer isDragging={isDragging}>
+          <Stack inline spacing={spacing}>
+            {React.Children.map(children, (child, idx) => {
+              if (!React.isValidElement(child)) return null;
+              return (
+                // eslint-disable-next-line no-return-assign
+                <StyledChild ref={r => (childRef[idx] = r && r.clientHeight)}>{child}</StyledChild>
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/HorizontalScroll/index.js:79]: what pattern do you suggest then?

[the remark this answers is not in the document]
Comment (silvenon) [reply packages/orbit-components/src/HorizontalScroll/index.js]: I think this should work:
```jsx
<StyledChild ref={r => {
  if (r) childRef[idx] = r.clientHeight
}}>
  {child}
</StyledChild>
```

File: packages/orbit-components/src/HorizontalScroll/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (silvenon) [reply packages/orbit-components/src/HorizontalScroll/README.md:26]: As a link it doesn't work, try it. It sounds like you thought I was suggesting changing the heading level.

[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/HorizontalScroll/README.md:26]: for me worked :thinking:

**Pull request merged**